yt_dlp/extractor/ciscolive.py

```diff
@@ -2,144 +2,179 @@
-    BRIGHTCOVE_URL_TEMPLATE = 'http://players.brightcove.net/5647924234001/SyK2FdqjM_default/index.html?videoId=%s'
+    _BASE_URL = 'https://www.ciscolive.com'
+    _TOKEN = None
 
-    HEADERS = {
-        'Origin': 'https://ciscolive.cisco.com',
-        'rfApiProfileId': RAINFOCUS_API_PROFILE_ID,
-        'rfWidgetId': RAINFOCUS_WIDGET_ID,
-    }
-
-    def _call_api(self, ep, rf_id, query, referrer, note=None):
-        headers = self.HEADERS.copy()
-        headers['Referer'] = referrer
-        return self._download_json(
-            self.RAINFOCUS_API_URL % ep, rf_id, note=note,
-            data=urlencode_postdata(query), headers=headers)
-
-    def _parse_rf_item(self, rf_item):
-        event_name = rf_item.get('eventName')
-        title = rf_item['title']
-        description = clean_html(rf_item.get('abstract'))
-        presenter_name = try_get(rf_item, lambda x: x['participants'][0]['fullName'])
-        bc_id = rf_item['videos'][0]['url']
-        bc_url = self.BRIGHTCOVE_URL_TEMPLATE % bc_id
-        duration = float_or_none(try_get(rf_item, lambda x: x['times'][0]['length']))
-        location = try_get(rf_item, lambda x: x['times'][0]['room'])
-
-        if duration:
-            duration = duration * 60
+    def _real_initialize(self):
+        self._TOKEN = try_call(lambda: self._get_cookies(self._BASE_URL)['rfjwt'].value)
+        if not self._TOKEN:
+            self.raise_login_required()
 
-        return {
-            '_type': 'url_transparent',
-            'url': bc_url,
-            'ie_key': 'BrightcoveNew',
-            'title': title,
-            'description': description,
-            'duration': duration,
-            'creator': presenter_name,
-            'location': location,
-            'series': event_name,
-        }
+    def _call_api(self, endpoint, some_id, payload=None, doseq=False, **kwargs):
+        if endpoint != 'search':
+            payload = {'id': some_id}
+
+        api_resp = self._download_json(
+            f'https://events.rainfocus.com/api/{endpoint}', some_id, headers={
+                'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+                'Rfapiprofileid': 'HEedDIRblcZk7Ld3KHm1T0VUtZog9eG9',
+                'Rfauthtoken': self._TOKEN,
+                'Rfwidgetid': 'M7n14I8sz0pklW1vybwVRdKrgdREj8sR',
+            }, data=urlencode_postdata(payload, doseq=doseq), **kwargs)
+        if traverse_obj(api_resp, 'responseCode') != '0':
+            msg = traverse_obj(api_resp, ('responseMessage', {str}, filter))
+            raise ExtractorError(
+                msg or 'API returned an error response', expected=bool(msg))
+
+        return api_resp
 
 
 class CiscoLiveSessionIE(CiscoLiveBaseIE):
-    _VALID_URL = r'https?://(?:www\.)?ciscolive(?:\.cisco)?\.com/[^#]*#/session/(?P<id>[^/?&]+)'
+    IE_NAME = 'ciscolive:session'
+    BRIGHTCOVE_URL_TEMPLATE = 'http://players.brightcove.net/5647924234001/SyK2FdqjM_default/index.html?videoId=%s'
```

Comment:
Really there should be a method `BrightcoveNew.make_url(cls, account_id, player_id, video_id, embed='default')`, and then `make_brightcove_url = functools.partial(BrightcoveNew.make_url, '5647924234001', 'SyK2FdqjM=')`.

But if not
```suggestion
    _BRIGHTCOVE_URL_TEMPLATE = 'http://players.brightcove.net/5647924234001/SyK2FdqjM_default/index.html?videoId=%s'
```
with the obvious consequential changes.

Reply:
I agree there should be a more concise method

I assumed the brightcove URL template was standard since it matches the other 21 examples, which is correct?

Reply:
agreed that there were some poor choices made with the brightcovenew extractor, e.g. account id always should have been included in the video id since `ref:` video ids are only unique per account.

but this is all very out-of-scope and doesn't belong in this PR